examples/apps/bookinfo/build-services.sh

```diff
@@ -27,19 +27,51 @@ if [ "$APP_VER" == "unknown" ]; then
   exit 1
 fi
 
+A8_TEST_ENV=$2
+
 # Remove the v from the version
 APP_VER=$(echo $APP_VER | sed "s/v//")
 
 SCRIPTDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 
+GENERATED_DOCKERFILE_NAME=Dockerfile.generated.sidecar
+A8_SIDECAR_SCRIPT_NAME=a8sidecar-testing.sh
+A8_SIDECAR_SCRIPT=$SCRIPTDIR/../../../testing/$A8_SIDECAR_SCRIPT_NAME
+A8_SIDECAR_TAR_NAME=a8sidecar-current.tar.gz
+A8_SIDECAR_TAR=$SCRIPTDIR/../../../release/$A8_SIDECAR_TAR_NAME
+
+generate_sidecar_dockerfile(){
+  if [ "$A8_TEST_ENV" == "examples" ]; then
+    cp $A8_SIDECAR_SCRIPT .
+    cp $A8_SIDECAR_TAR .
+    sed 's#{COPY_SCRIPT}#COPY '"$A8_SIDECAR_SCRIPT_NAME"' /opt/microservices/#; s#{COPY_SIDECAR_TAR}#COPY '"$A8_SIDECAR_TAR_NAME"' /opt/microservices/#; s#RUN wget -qO- https://github.com/amalgam8/amalgam8/releases/download/{A8_RELEASE}/a8sidecar.sh | sh#'"RUN /opt/microservices/$A8_SIDECAR_SCRIPT_NAME"'#' Dockerfile.sidecar > $GENERATED_DOCKERFILE_NAME
+  else
+    sed 's/{COPY_SCRIPT}//; s/{COPY_SIDECAR_TAR}//; s/{A8_RELEASE}/'"v$APP_VER"'/' Dockerfile.sidecar > $GENERATED_DOCKERFILE_NAME
+  fi
+}
+
+cleanup_generated_files(){
+  if [ "$A8_TEST_ENV" == "examples" ]; then
+    rm $GENERATED_DOCKERFILE_NAME $A8_SIDECAR_SCRIPT_NAME $A8_SIDECAR_TAR_NAME
+  else
+    rm $GENERATED_DOCKERFILE_NAME
+  fi
+}
+
+echo "Building productpage"
 pushd $SCRIPTDIR/productpage
   docker build -t amalgam8/a8-examples-bookinfo-productpage-v1:$APP_VER .
-  docker build -t amalgam8/a8-examples-bookinfo-productpage-sidecar-v1:$APP_VER -f Dockerfile.sidecar .
+  generate_sidecar_dockerfile
+  docker build -t amalgam8/a8-examples-bookinfo-productpage-sidecar-v1:$APP_VER -f $GENERATED_DOCKERFILE_NAME .
+  cleanup_generated_files
 popd
 
+echo "Building details"
 pushd $SCRIPTDIR/details
   docker build -t amalgam8/a8-examples-bookinfo-details-v1:$APP_VER .
-  docker build -t amalgam8/a8-examples-bookinfo-details-sidecar-v1:$APP_VER -f Dockerfile.sidecar .
+  generate_sidecar_dockerfile
```

Comment:
Wouldn't it be simpler to templatize the example docker files and run the " generate dockerfile app name" command?